RFC-0003-arrow-flight-connector.md

```diff
@@ -0,0 +1,83 @@
+# **RFC0003 for Presto**
+
+## Arrow Flight connector template in Presto
+
+Proposers
+
+* Elbin Pallimalil
+
+## [Related Issues]
+
+https://github.com/prestodb/presto/issues/22729
+
+## Summary
+
+Create a base Arrow Flight module in Presto which can be extended to create an Arrow Flight connector.
+
+## Background
+
+Apache Arrow Flight is a new general-purpose client-server framework that simplifies high performance transport of large datasets over network interfaces. https://arrow.apache.org/blog/2019/10/13/introducing-arrow-flight/
+
+Presto currently supports different datasources through connectors for each type of datasource. The connectors can use JDBC or some other protocol for datasource connection. If we use Flight client to connect to the different datasources we can improve the performance of the datasource connection. Also we would only need to develop one Flight client that can be used across different datasources instead of developing individual connectors for each type of datasource.
```

Comment:
We have individual connectors for each type of datasource like MySQL, Oracle or PostgresQL. If we develop one Flight connector, it can be used across different types of datasources.

Reply:
> Also we would only need to develop one Flight client that can be used across different datasources instead of developing individual connectors for each type of datasource.

This statement is confusing. Isn't Flight client similar to JDBC in terms of capabilities? We have different connectors on top of the JDBC standard because each data source requires customizations.

Reply:
I meant that we don't have to create different connectors for each type of datasource like we do in JDBC.

Reply:
Why do you think is missing in JDBC that arrow Flight solves?

Reply:
Arrow Flight provides better performance than JDBC. The Arrow memory format supports zero-copy reads for lightning-fast data access without serialization overhead.